PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.Application/Services/Doctor/DoctorService.cs

```diff
@@ -0,0 +1,106 @@
+using FluentValidation;
+using FluentValidation.Results;
+using PrivateHospitals.Application.Dtos.Doctor;
+using PrivateHospitals.Application.Dtos.Patient;
+using PrivateHospitals.Application.Dtos.User;
+using PrivateHospitals.Application.Interfaces.Doctor;
+using PrivateHospitals.Application.Responses;
+using PrivateHospitals.Core.Models;
+using PrivateHospitals.Data.Interfaces.Doctor;
+using PrivateHospitals.Data.Interfaces.Patient;
+using PrivateHospitals.Data.Interfaces.User;
+
+namespace PrivateHospitals.Application.Services.Doctor;
+
+public class DoctorService: IDoctorService
+{
+    private readonly IDoctorRepository _doctorRepository;
+    private readonly IValidator<RegisterDto> _registerDtoValidator;
+    private readonly IValidator<RegisterDoctorDto> _registerDoctorDtoValidator;
```

Comment:
Better to do validation in AspNet middleware pipeline. Either use built in functionality which is deprecated, or I recommend to create a custom middleware for validation - https://youtu.be/OhQA4PDdp0Q?t=568

Reply:
Fixed